src/libraries/System.Formats.Nrbf/ref/System.Formats.Nrbf.csproj

```diff
@@ -0,0 +1,20 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <TargetFrameworks>$(NetCoreAppCurrent);$(NetCoreAppPrevious);$(NetCoreAppMinimum);netstandard2.0;$(NetFrameworkMinimum)</TargetFrameworks>
+    <CLSCompliant>false</CLSCompliant>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <Compile Include="System.Formats.Nrbf.cs" />
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(TargetFramework)' == '$(NetCoreAppCurrent)'">
+    <ProjectReference Include="$(LibrariesProjectRoot)System.IO.Hashing\ref\System.IO.Hashing.csproj" />
+    <ProjectReference Include="$(LibrariesProjectRoot)System.Reflection.Metadata\ref\System.Reflection.Metadata.csproj" />
```

Comment:
@ViktorHofer @carlossanlop @ericstj  despite adding the reference to System.Reflection.Metadata ref project, I keep getting following errors when performing the build of the solution:

```cmd
.\dotnet.cmd build .\src\libraries\System.Formats.Nrbf\System.Formats.Nrbf.sln -c Release
```

```log
D:\projects\runtime\src\libraries\System.Formats.Nrbf\ref\System.Formats.Nrbf.cs(56,34): error CS0234: The type or namespace name 'Metadata' does not exist in the namespace 'System.Reflection' (are you missing an assembly reference?) [
D:\projects\runtime\src\libraries\System.Formats.Nrbf\ref\System.Formats.Nrbf.csproj::TargetFramework=netstandard2.0]
D:\projects\runtime\src\libraries\System.Formats.Nrbf\ref\System.Formats.Nrbf.cs(75,43): error CS0234: The type or namespace name 'Metadata' does not exist in the namespace 'System.Reflection' (are you missing an assembly reference?) [
D:\projects\runtime\src\libraries\System.Formats.Nrbf\ref\System.Formats.Nrbf.csproj::TargetFramework=netstandard2.0]
D:\projects\runtime\src\libraries\System.Formats.Nrbf\ref\System.Formats.Nrbf.cs(22,43): error CS0234: The type or namespace name 'Metadata' does not exist in the namespace 'System.Reflection' (are you missing an assembly reference?) [
D:\projects\runtime\src\libraries\System.Formats.Nrbf\ref\System.Formats.Nrbf.csproj::TargetFramework=netstandard2.0]
D:\projects\runtime\src\libraries\System.Formats.Nrbf\ref\System.Formats.Nrbf.cs(75,43): error CS0234: The type or namespace name 'Metadata' does not exist in the namespace 'System.Reflection' (are you missing an assembly reference?) [
D:\projects\runtime\src\libraries\System.Formats.Nrbf\ref\System.Formats.Nrbf.csproj::TargetFramework=net462]
D:\projects\runtime\src\libraries\System.Formats.Nrbf\ref\System.Formats.Nrbf.cs(67,43): error CS0234: The type or namespace name 'Metadata' does not exist in the namespace 'System.Reflection' (are you missing an assembly reference?) [
D:\projects\runtime\src\libraries\System.Formats.Nrbf\ref\System.Formats.Nrbf.csproj::TargetFramework=netstandard2.0]
D:\projects\runtime\src\libraries\System.Formats.Nrbf\ref\System.Formats.Nrbf.cs(22,43): error CS0234: The type or namespace name 'Metadata' does not exist in the namespace 'System.Reflection' (are you missing an assembly reference?) [
D:\projects\runtime\src\libraries\System.Formats.Nrbf\ref\System.Formats.Nrbf.csproj::TargetFramework=net462]
D:\projects\runtime\src\libraries\System.Formats.Nrbf\ref\System.Formats.Nrbf.cs(56,34): error CS0234: The type or namespace name 'Metadata' does not exist in the namespace 'System.Reflection' (are you missing an assembly reference?) [
D:\projects\runtime\src\libraries\System.Formats.Nrbf\ref\System.Formats.Nrbf.csproj::TargetFramework=net462]
D:\projects\runtime\src\libraries\System.Formats.Nrbf\ref\System.Formats.Nrbf.cs(67,43): error CS0234: The type or namespace name 'Metadata' does not exist in the namespace 'System.Reflection' (are you missing an assembly reference?) [
D:\projects\runtime\src\libraries\System.Formats.Nrbf\ref\System.Formats.Nrbf.csproj::TargetFramework=net462]
D:\projects\runtime\src\libraries\System.Formats.Nrbf\ref\System.Formats.Nrbf.cs(75,52): error CS0234: The type or namespace name 'TypeName' does not exist in the namespace 'System.Reflection.Metadata' (are you missing an assembly refe
rence?) [D:\projects\runtime\src\libraries\System.Formats.Nrbf\ref\System.Formats.Nrbf.csproj::TargetFramework=net8.0]
D:\projects\runtime\src\libraries\System.Formats.Nrbf\ref\System.Formats.Nrbf.cs(56,43): error CS0234: The type or namespace name 'TypeNameParseOptions' does not exist in the namespace 'System.Reflection.Metadata' (are you missing an a
ssembly reference?) [D:\projects\runtime\src\libraries\System.Formats.Nrbf\ref\System.Formats.Nrbf.csproj::TargetFramework=net8.0]
D:\projects\runtime\src\libraries\System.Formats.Nrbf\ref\System.Formats.Nrbf.cs(22,52): error CS0234: The type or namespace name 'TypeName' does not exist in the namespace 'System.Reflection.Metadata' (are you missing an assembly refe
rence?) [D:\projects\runtime\src\libraries\System.Formats.Nrbf\ref\System.Formats.Nrbf.csproj::TargetFramework=net8.0]
D:\projects\runtime\src\libraries\System.Formats.Nrbf\ref\System.Formats.Nrbf.cs(67,52): error CS0234: The type or namespace name 'TypeName' does not exist in the namespace 'System.Reflection.Metadata' (are you missing an assembly refe
rence?) [D:\projects\runtime\src\libraries\System.Formats.Nrbf\ref\System.Formats.Nrbf.csproj::TargetFramework=net8.0]
```

A full build is passing fine:

```cmd
.\build.cmd -c Release -subset clr+libs+libs.tests
```

But CI is red. Any ideas what can be wrong?

Reply:
Answered with a suggestion https://github.com/dotnet/runtime/pull/103232/files#r1635294815

Reply:
Full build passes fine because that only builds .NETCore for the platform you're on - a *vertical* build.  

The thing failing here is the multitargeted build - which is the normal thing you get with a dotnet build or pack.

As Carlos suggested, it's because you have incomplete projectreferences on other frameworks.  Make sure you have project references necessary for every framework you build for.  My personal preference is to include direct references for every assembly I require.

Reply:
Please excuse me for bothering you, I must have been really tired yesterday to not see the fact that I am referencing the right projects only for one TFM.

Thanks for your help!